Update tar split

dmcgowan opened the pull request:
Pulls in fix for tar split disassembly incorrectly storing the padding at the end of the file, leading to a hash difference on the re-assembled tar archive. Note that the original error only occurs when the tar does not contain the typical number of null byte blocks at the end of the tar.

Since the error occurred on disassembly of the tar, images which were effected by this bug must be removed and reloaded/re-pulled. 

Closes #26434

Ping @aaronlehmann 
Ping @vbatts ( I can update this to a tag if you create one and prefer to use that )

aaronlehmann commented:
Thanks.

I see there are quite a few changes in tar-split. Do you feel comfortable including this in 1.12.2?

vbatts commented:
@dmcgowan i've tagged this as v0.10.1 for easier reference

vbatts commented:
@aaronlehmann there were changes, but it's mostly the rebase on upstream `archive/tar` from go1.6. It passes all the existing tests, and now fixes additional issues that @dmcgowan found. I'll let derek decide for you 1.12.2 release though.

dmcgowan committed:
Update tar split

Pulls in fix for tar split disassembly incorrectly storing the padding
at the end of the file, leading to a hash difference on the
re-assembled tar archive.

Closes #26434

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

dmcgowan commented:
Updated to use tag. I would feel more comfortable skipping for 1.10.2 since it is coming up pretty quickly. So far the impact of the bug seems pretty minimal and appears to only effect tars which were corrupted or not generated by a docker push/save operation. It is important for tar-split to be reliable and exact but I don't see the urgency in rushing it into the next point release.

aaronlehmann commented:
I'm not sure either way. The problem is there's no workaround to "migrate" an image. You have to delete it and start over. That's a pretty bad situation.

tonistiigi commented:
Is there a possibility to vendor only the fix part of it for `v1.12.2`? Given that we already missed `v1.12.2-rc1` this seems too much of a risk to me.

LGTM for master

dmcgowan commented:
@tonistiigi that would require further investigation as the bug that was fixed from the 0.9.13 tag did seem to exhibit the same behavior as the original bug with vendored 0.9.11 tag. It is possible the fix is the same if we wanted to branch 0.9.11 for the Docker 1.12 branch.

LK4D4 commented:
LGTM for master
I suppose it's too late for 1.12.2
ping @vieux
